extensions/pythonprocessors/h2o/h2o3/mojo/ExecuteH2oMojoScoring.py

```diff
@@ -0,0 +1,165 @@
+
+    sudo apt-get -y update
+
+    Install Java to include open source H2O-3 algorithms:
+    
+    sudo apt-get -y install openjdk-8-jdk
+
+    Install Datatable and pandas:
+
+    pip install datatable
+    pip install pandas
+
+    Option 1: Install H2O-3 with conda
+
+    conda create -n h2o3-nifi-minifi python=3.6
+    conda activate h2o3-nifi-minifi
+    conda config --append channels conda-forge
+    conda install -y -c h2oai h2o
+
+    Option 2: Install H2O-3 with pip
+
+    pip install requests
+    pip install tabulate
+    pip install "colorama>=0.3.8"
+    pip install future
+    pip uninstall h2o
+    If on Mac OS X, must include --user:
+        pip install -f http://h2o-release.s3.amazonaws.com/h2o/latest_stable_Py.html h2o --user
+    else:
+        pip install -f http://h2o-release.s3.amazonaws.com/h2o/latest_stable_Py.html h2o
+
+"""
+import h2o
+import codecs
+import pandas as pd
+import datatable as dt
+
+mojo_model = None
+
+def describe(processor):
+    """ describe what this processor does
+    """
+    processor.setDescription("Executes H2O-3's MOJO Model in Python to do batch scoring or \
+        real-time scoring for one or more predicted label(s) on the tabular test data in \
+        the incoming flow file content. If tabular data is one row, then MOJO does real-time \
+        scoring. If tabular data is multiple rows, then MOJO does batch scoring.")
+
+def onInitialize(processor):
+    """ onInitialize is where you can set properties
+        processor.addProperty(name, description, defaultValue, required, el)
+    """
+    processor.addProperty("MOJO Model Filepath", "Add the filepath to the MOJO Model file. For example, \
+        'path/to/mojo-model/GBM_grid__1_AutoML_20200511_075150_model_180.zip'.", "", True, False)
+
+    processor.addProperty("Is First Line Header", "Add True or False for whether first line is header.", \
+        "True", True, False)
+
+    processor.addProperty("Input Schema", "If first line is not header, then you must add Input Schema for \
+        incoming data.If there is more than one column name, write a comma separated list of \
+        column names. Else, you do not need to add an Input Schema.", "", False, False)
+
+    processor.addProperty("Use Output Header", "Add True or False for whether you want to use an output \
+        for your predictions.", "False", False, False)
+
+    processor.addProperty("Output Schema", "To set Output Schema, 'Use Output Header' must be set to 'True' \
+        If you want more descriptive column names for your predictions, then add an Output Schema. If there \
+        is more than one column name, write a comma separated list of column names. Else, H2O-3 will include \
+        them by default", "", False, False)
+
+def onSchedule(context):
+    """ onSchedule is where you load and read properties
+        this function is called 1 time when the processor is scheduled to run
+    """
+    global mojo_model
```

Comment:
Is it the case that this applies to all processors instances that may be using this?

Reply:
I use global for mojo_model, so I can access mojo_model in the onSchdule() function and onTrigger() function. In onSchedule(), I specify global since the mojo_model will change in this function and my intention is to instantiate a mojo_model object 1 time right at the start when the processor is scheduled to run. This case applies to all processor instances. Then in the onTrigger(), we use the mojo_model to make predictions. 

For example, one processor instance could instantiate a classification mojo_model while another processor could instantiate a regression mojo_model. Then we have one processor that is making classification predictions while the other processor is making regression predictions.

Would each processor has its own unique mojo_model object or no since I am using global?